registrations/tasks.py

```diff
@@ -1,23 +1,225 @@
+import datetime
+import six  # for Python 2 and 3 string type compatibility
+
 from celery.task import Task
 from celery.utils.log import get_task_logger
 
+from .models import Registration
+
 
 logger = get_task_logger(__name__)
 
 
-class The_Incr(Task):
+def get_today():
+    return datetime.today()
+
+
+def is_valid_date(date):
+    try:
+        datetime.datetime.strptime(date, "%Y%m%d")
+        return True
+    except:
+        return False
+
+
+def is_valid_uuid(id):
+    return len(id) == 36 and id[14] == '4' and id[19] in ['a', 'b', '8', '9']
+
+
+def is_valid_lang(lang):
+    return lang in ["english", "runyakore", "lusoga"]
+
+
+def is_valid_msg_type(msg_type):
+    return msg_type in ["sms"]  # currently sms-only
+
+
+def is_valid_msg_receiver(msg_receiver):
+    return msg_receiver in ["head_of_household", "mother_to_be",
+                            "family_member", "trusted_friend"]
+
+
+def is_valid_loss_reason(loss_reason):
+    return loss_reason in ['miscarriage', 'stillborn', 'baby_died']
 
+
+def is_valid_name(name):
+    return isinstance(name, six.string_types)  # TODO reject non-letters
+
+
+def is_valid_id_type(id_type):
+    return id_type in ["ugandan_id", "other"]
+
+
+def is_valid_id_no(id_no):
+    return isinstance(id_no, six.string_types)  # TODO proper id validation
+
+
+def calc_pregnancy_week_lmp(today, lmp):
+    """ Calculate how far along the mother's prenancy is in weeks.
     """
-    Task to incr something
+    last_period_date = datetime.datetime.strptime(lmp, "%Y%m%d")
+    time_diff = today - last_period_date
+    preg_weeks = int(time_diff.days / 7)
+    # You can't be one week pregnant (smaller numbers will be rejected)
+    if preg_weeks == 1:
+        preg_weeks = 2
+    return preg_weeks
+
+
+class ValidateRegistration(Task):
+    """ Task to validate a registration model entry's registration
+    data.
     """
-    name = "familyconnect_registration.registrations.tasks.the_incr"
+    name = "familyconnect_registration.registrations.tasks.\
+    validate_registration"
 
-    def run(self, anum, **kwargs):
+    def check_field_values(self, fields, registration_data):
+        for field in fields:
+            if field in ["contact", "registered_by"]:
+                if not is_valid_uuid(registration_data[field]):
```

Comment:
Not sure there's much point in doing this... A useful validation would be to actually validate it against the identity store.